app/src/main/java/io/gnosis/safe/ui/safe/terms/TermsBottomSheetDialog.kt

```diff
@@ -0,0 +1,63 @@
+package io.gnosis.safe.ui.safe.terms
+
+import android.text.Html
+import android.text.method.LinkMovementMethod
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.Observer
+import com.google.android.material.bottomsheet.BottomSheetDialog
+import io.gnosis.safe.HeimdallApplication
+import io.gnosis.safe.R
+import io.gnosis.safe.di.components.DaggerViewComponent
+import io.gnosis.safe.di.modules.ViewModule
+import kotlinx.android.synthetic.main.bottom_sheet_terms_and_conditions.*
+import javax.inject.Inject
+
+class TermsBottomSheetDialog(val activity: AppCompatActivity) : BottomSheetDialog(activity) {
+
+    @Inject
+    lateinit var viewModel: TermsViewModel
+
+    init {
+        inject()
+
+        setContentView(layoutInflater.inflate(R.layout.bottom_sheet_terms_and_conditions, null))
+
+        bottom_sheet_terms_and_conditions_privacy_policy_link.text = Html.fromHtml(activity.getString(R.string.terms_privacy_link))
+        bottom_sheet_terms_and_conditions_privacy_policy_link.movementMethod = LinkMovementMethod.getInstance()
+
+        bottom_sheet_terms_and_conditions_terms_of_use_link.text = Html.fromHtml(activity.getString(R.string.terms_terms_of_use_link))
+        bottom_sheet_terms_and_conditions_terms_of_use_link.movementMethod = LinkMovementMethod.getInstance()
+
+        bottom_sheet_terms_and_conditions_agree.setOnClickListener {
+            onAgree()
+        }
+
+        bottom_sheet_terms_and_conditions_reject.setOnClickListener {
+            onReject()
+        }
+
+        viewModel.show.observe(activity, Observer { show ->
+            if (show) {
+                this.show()
+            }
+        })
+    }
+
+    private fun onAgree() {
+        dismiss()
+        viewModel.onAgreeClicked()
+    }
+
+    private fun onReject() {
+        dismiss()
+    }
+
+    private fun inject() {
+        DaggerViewComponent.builder()
+            .applicationComponent(HeimdallApplication[context])
+            .viewModule(ViewModule(context))
+            .build().inject(this)
+    }
+
+    fun checkTerms(function: () -> Unit) = viewModel.checkTerms { function() }
```

Comment:
Hmm so here you are passing the context from the SplashActivity, to the BottomSheetDialog and then into the viewModel, and as you are storing a reference to the method, you still have a reference to a view in the viewModel. I would suggest looking at the `EnsInputDialog`. If you reverse the direction of the communication between classes, you won't have this problem.

The viewModel notifies that the terms have been agreed to, the dialog triggers an `onAgreed` callback to which it holds a reference and then the calling context (SplashActivity) gets notified.